### ISSUE TITLE
`this.props` undefined on the unstable release

### ISSUE THREAD

**Issue by johnvonneumann7** [iamhosseindhv/notistack#393]
When I use notistack alpha in material-ui v5, I get the error `Cannot read property 'props' of undefined.`

If I change the notistack version to `1.0.6`, I don't get this error.
Is there anything I can do to help?

| Tech         | Version |
|--------------|---------|
| Notistack | 2.0.1-alpha.4  |
| material-ui | 5.0.0-alpha.34 |

**Comment by Moshyfawn** [comment]
Same here. I've upgraded from `@material-ui/core: 5.0.0-alpha.34` to `@material-ui/core: 5.0.0-alpha.36` since there're a couple of changes I really needed to test out. It lead to breaking changes described in #395. So I tried upgrading `notistack: 1.0.9` to `notistack: 2.0.1-alpha.4` but got the `Can't access property "props", this is undefined`.
It breaks on `const merger = merge(options, this.props, DEFAULTS);` trying to merge Provider's props with the individual snack's options.
The `SnackbarProvider` is setup within a MUI `ThemeProvider`

**Comment by shahalihridoy** [comment]
any update?

**Comment by vdv73rus** [comment]
Maybe merge #397 ?

**Comment by kkaplita** [comment]
Any news about this?

**Comment by johnvonneumann7** [comment]
I have yet to solve this problem.

**Comment by kkaplita** [comment]
> I have yet to solve this problem.

Do you have an approximate date when it will be fixed?

**Comment by bpred754** [comment]
This is blocking me as well after upgrading Material UI

**Comment by bpred754** [comment]
Someone has put together a fork of this library with the fix. You can find it here: https://www.npmjs.com/package/notistack5

[Original Comment](https://github.com/iamhosseindhv/notistack/issues/395#issuecomment-872046905)

**Comment by iamhosseindhv** [comment]
Apologies for responding late. Notistack `alpha` (a.k.a v2) is not ready to be used yet as it's still in the development stage. I'll fix this specific issue on `alpha` and publish a new package anyway.

I appreciate it's quite easy to mistake it with `next` which is probably the one you're looking for. `next` branch is supposed to be compatible with MUI v5 and we have #387 to make that happen.

> @Moshyfawn  It breaks on `const merger = merge(options, this.props, DEFAULTS);`, trying to merge Provider's props with the individual snack's options.

I don't see how this is an issue when the same line of code is in `master` branch and already published in the latest release.

So in other words if you're using:
* Unstable MUI `>= 5.0.0-alpha.35`:  Wait for #387 to be merged.
* MUI `<= 5.0.0-alpha.34` or `>= 4.12.0`:  Use `npm i notistack@next` or lock to version `npm i notistack@1.0.6-next.1`. See a working [example](https://codesandbox.io/s/epic-star-dzx7k?file=/App.js)
* MUI `< 4.12.0`: Use the latest. `npm i notistack` or `npm i notistack@latest`.

I appreciate this is a mess at the moment. The whole point of v2 (which I'm working on) is to remove the dependency on MUI + lots of customisation choices.

**Comment by luminaxster** [comment]
If the error is related to  this line:
https://github.com/iamhosseindhv/notistack/blob/54acc22077ca58234295276f6ed6d4940353cc01/src/SnackbarProvider.tsx#L59

I think you have some unbound methods that are offered as functions (this persists in next and alpha branches):
https://github.com/iamhosseindhv/notistack/blob/54acc22077ca58234295276f6ed6d4940353cc01/src/SnackbarProvider.tsx#L35

Bind them before exposing them:
```js
const enqueueSnackbar = this.enqueueSnackbar.bind(this);
const closeSnackbar = this.closeSnackbar.bind(this);
//...
contextValue: {
                enqueueSnackbar,
                closeSnackbar,
            },
```
I'm not sure if you have some auto bind conf in your bundler, but correcting this in the debugger fixes it.

**Comment by iamhosseindhv** [comment]
Fixed in #397 . Fix is published as `v2.0.1-alpha.5`, which can be downloaded by `npm i notistack@alpha`.
